src/test.js

```diff
@@ -659,8 +692,7 @@ export function internalStop( test ) {
 	if ( config.testTimeout && defined.setTimeout ) {
 		clearTimeout( config.timeout );
 		config.timeout = setTimeout( function() {
-			pushFailure( "Test timed out", sourceFromStacktrace( 2 ) );
-			internalRecover( test );
+			pushFailureAndResume( "Test timed out", sourceFromStacktrace( 2 ) );
```

Comment:
Hmm. If we already have a `test` variable, I wonder if I should instead do `test.pushFailure(/* args */)` followed by `internalRecover( test )`? Somehow I didn't think of this until after I pushed the branch.